x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -0,0 +1,243 @@
+          // pass this back with the privileges to allow for the disabling of UI controls.
+          upgradeInProgress = info.upgrade_mode === true;
+        } catch (error) {
+          // if the ml.info check fails, it could be due to the user having insufficient privileges
+          // most likely they do not have the ml_user role and therefore will be blocked from using
+          // ML at all. However, we need to catch this error so the privilege check doesn't fail.
+          if (error.status === 403) {
+            mlLog.info(
+              'Unable to determine whether upgrade is being performed due to insufficient user privileges'
+            );
+          } else {
+            mlLog.warn('Unable to determine whether upgrade is being performed');
+          }
+        }
+
+        if (isSecurityDisabled(xpackMainPlugin)) {
+          // if xpack.security.enabled has been explicitly set to false
+          // return that security is disabled and don't call the privilegeCheck endpoint
+          return response.ok({
+            body: {
+              securityDisabled: true,
+              upgradeInProgress,
+            },
+          });
+        } else {
+          const body = request.body;
+          const resp = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', { body });
+          resp.upgradeInProgress = upgradeInProgress;
+          return response.ok({
+            body: resp,
+          });
+        }
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/ml_capabilities Check ML capabilities
+   * @apiName MlCapabilities
+   * @apiDescription Checks ML capabilities
+   */
+  router.get(
+    {
+      path: '/api/ml/ml_capabilities',
+      validate: {
+        query: schema.object({
+          ignoreSpace: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const ignoreSpaces = request.query && request.query.ignoreSpaces === 'true';
+        // if spaces is disabled force isMlEnabledInSpace to be true
+        const { isMlEnabledInSpace } =
+          spacesPlugin !== undefined
+            ? spacesUtilsProvider(spacesPlugin, (request as unknown) as Request)
+            : { isMlEnabledInSpace: async () => true };
+
+        const { getPrivileges } = privilegesProvider(
+          context.ml!.mlClient.callAsCurrentUser,
+          xpackMainPlugin,
+          isMlEnabledInSpace,
+          ignoreSpaces
+        );
+        return response.ok({
+          body: await getPrivileges(),
+        });
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/ml_node_count Get the amount of ML nodes
```

Comment:
Should be `get`

Reply:
https://github.com/elastic/kibana/pull/57835/commits/608f48dbacc75c6842b84eac804650de03190c1b